packages/apps/src/app.ts

```diff
@@ -243,15 +250,15 @@ export class App {
     }
 
     // add/validate plugins
-    const plugins = this.options.plugins || [];
+    const plugins: Array<TPlugin> = this.options.plugins || [];
     let httpPlugin = plugins.find((p) => {
       const meta = getMetadata(p);
       return meta.name === 'http';
     }) as HttpPlugin | undefined;
 
     if (!httpPlugin) {
       httpPlugin = new HttpPlugin();
-      plugins.unshift(httpPlugin);
+      plugins.unshift(httpPlugin as any);
```

Comment:
is the `as any` avoidable?

Reply:
It might be, with some level of annoying-effort. I'll need to see...

Reply:
if it's too much of a hassle then i think it's fine. just don't want to make this a common practice.

Reply:
it might be worth throwing an warning if two plugins that support the same events are imported to an app, what do you guys think?

Reply:
So right now, the events are a type-level check. The only place we do a runtime check is in apps.plugins. What it does there there is just ensures that emissions for core-level events are not take-over by a plugin. I'll warn there.

Reply:
I did some work to see if any was avoidable, but it's a stupid amount of indirection to do it, so I'm just leaning toward keeping it like it is for now. Lmk if y'all disagree